app/_src/gateway/production/deployment-topologies/hybrid-mode/hybrid-mode-spec.md

```diff
@@ -0,0 +1,114 @@
+
+```
+{ "type": "reconfigure",
+  "config_table": {
+    "_format_version": "1.1",
+    "consumers": [
+      {
+        "username": "user",
+        "tags": [
+          "internal-user"
+        ],
+        "plugins": [
+          {
+            "name": "rate-limiting",
+            "enabled": true,
+            "protocols": [
+              "http",
+              "https"
+            ]
+          }
+        ]
+      }
+    ]
+  }
+}
+```
+
+
+All frames transferred from the CP to the DP are Gziped before being sent over WebSocket. This reduces the size of the payload significantly since Kong configs are highly compressible.
+
+### Sensitive data management 
+
+Because the CP node dumps the entire database config and send it to the DP, the following data may be included: 
+
+* Private key of TLS certificate for proxied domains.
+* IP address, port, client certificates, private keys and any other information used to access upstream services.
+* Consumer information including emails, authentication plugin credentials like an API Key for the key-auth plugin.
+* Unique ID for identifying the hybrid mode cluster. This is generated once and shared by every CP and DP within the same cluster. This ID is used for anonymous reporting and if anonymous reporting is not disabled.
+
+The DP nodes also store the latest received configuration data on the filesystem in order to survive CP failures or streamline DP startup.
+
+## Heartbeat and version reporting
+
+The CP needs to know what the config version of each DP that is running in order to know if the DP is up to date, or if a config has failed. The config version is reported in the WebSocket Ping frame that the DP generates every 30 seconds. If a new config was pushed and applied successfully, then a Ping frame is generated immediately by the DP in order to report the version change to the CP. The Ping frame is also used to keep the TCP connection open so that configs can be pushed immediately. 
+
+
+## FAQ
+
+### How long can a disconnected DP continue working without impact to operations? 
+
+Indefinitely. Because DP nodes cache the last known config from the CP, they will continue to function with that config indefinitely, even after restarts. However, there are exceptions that are worth noting. 
+
+1. If the EE license expires the DP may stop working. 
+2. If the config cache on the DP node is deleted, then the next time it restarts the DP will not have any configuration applied to it. 
+3. Provisioning a new DP is not possible if the CP is down. However, the user could choose to use the `KONG_DECLARATIVE_CONFIG` option to load a fallback YAML config if the hybrid mode cache is unavailable. 
+
+### Can a disconnected DP be restarted and still reconfigure itself based on the previous snapshot? 
+
+Yes. 
+
+
+### Can the DP configuration change in disconnected mode?
```

Comment:
In 3.x we use LMDB to store config so this method can not be used anymore.

For enterprise version, we have Data Plane Resilience so data planes can load config from an S3 compatible bucket.